frontend/src/Reducer-context.js

```diff
@@ -0,0 +1,204 @@
+          showInventoryAll,
+          showMaterial,
+          showIdealFor,
+          showDiscount,
+          showNew,
+          cartlist,
+          route,
+          wishlist,
+          dispatch
+        }}
+      >
+        {children}
+      </Reducercontext.Provider>
+    </>
+  );
+}
+
+export function useReduce() {
+  return useContext(Reducercontext);
+}
+
+function reducer(state, action) {
+  //console.log("reducer fn");
+
+  switch (action.type) {
+    case "ROUTE":
+      return { ...state, route: action.payload };
+    case "IDEAL":
+      return { ...state, showIdealFor: action.payload };
+    case "MATERIAL":
+      return { ...state, showMaterial: action.payload };
+    case "SORT":
+      return {
+        ...state,
+        sortBy: action.payload
+      };
+    case "NEW":
+      return {
+        ...state,
+        showNew: action.payload
+      };
+    case "OFFER":
+      return {
+        ...state,
+        showDiscount: action.payload
+      };
+    case "TOGGLE_INVENTORY":
+      return {
+        ...state,
+        showInventoryAll: !state.showInventoryAll
+      };
+
+    case "TOGGLE_DELIVERY":
+      return { ...state, showFastDeliveryOnly: !state.showFastDeliveryOnly };
+
+    case "ADD_TO_CART":
+      return { ...state, cartlist: [...state.cartlist, action.payload] };
+
+    case "INCREASE_IN_CART":
+      return {
+        ...state,
+        cartlist: state.cartlist.map((item) => {
+          if (item.itemId === action.payload) {
+            return { ...item, qty: item.qty + 1 };
+          }
+          return item;
+        })
+      };
+
+    case "DECREASE_IN_CART":
+      if (
+        state.cartlist.find((item) => item.itemId === action.payload).qty === 1
+      ) {
+        console.log("in del");
+        return {
+          ...state,
+          cartlist: state.cartlist.filter(
+            (item) => item.itemId !== action.payload
+          )
+        };
+      }
+
+      return {
+        ...state,
+        cartlist: state.cartlist.map((item) => {
+          if (item.itemId === action.payload) {
+            return { ...item, qty: item.qty - 1 };
+          }
+          return item;
+        })
+      };
+
+    case "REMOVE_FROM_CARTLIST":
+      return {
+        ...state,
+        cartlist: state.cartlist.filter(
+          (item) => item.itemId !== action.payload
+        )
+      };
+
+    case "ADD_WISH":
+      return { ...state, wishlist: [...state.wishlist, action.payload] };
+
+    case "REMOVE_WISH":
+      return {
+        ...state,
+        wishlist: state.wishlist.filter((item) => item !== action.payload)
+      };
+
+    case "LOAD_CART":
+      return {
+        ...state,
+        cartlist: action.payload
+      };
+
+    case "LOAD_WISHLIST":
+      return {
+        ...state,
+        wishlist: action.payload
+      };
+    default:
+      return console.log("error");
+  }
+}
+
+async function deleting(url, payload) {
+  return await axios.delete(url, payload);
+}
```

Comment:
Looks like this code isn't being used, if so please remove it

Reply:
Thanks for pointing this out. I completely missed this part. I will remove this part of code.